sdk/code-analysis/SharedAnalysisCode/AbstractNullCollectionInitializerAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Data;
+using System.Collections.Immutable;
+
+namespace Amazon.CodeAnalysis.Shared
+{
+    public abstract class AbstractNullCollectionInitializerAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "AWSSDK2000";
```

Comment:
My thought process is we could reserve AWSSDK2XXX for any code analysis rules dealing with nullability.